[decomps] Add decomposition for linalg_vector_norm

SS-JIA opened the pull request:
Summary:
## Context

`linalg_vector_norm` is required to run Vision Transformer models in ExecuTorch. Currently, model export + inference fails because ExecuTorch doesn't have a kernel for `linalg_vector_norm`.

However, there is a decomposition for the operator. Add the decomposition to the core decomp table to unblock ExecuTorch.

a-holm commented:
Looking at the diff, it seems the actual code for the decomposition might be missing. The current changes only show the removal of a blank line in `torch/_decomp/decompositions.py`, which doesn't align with the goal described in the PR title and description.

angelayi reviewed (changes requested):
Adding `aten.linalg_vector_norm` to [this list](https://github.com/pytorch/pytorch/blob/5e34758cef85b497a6ba313cc09c4cac283583d5/torch/_decomp/__init__.py#L295) fixes the issue

SS-JIA commented:
Updated with the appropriate change. PTAL @angelayi

Commit:
[decomps] Add decomposition for linalg_vector_norm

Summary:
## Context

`linalg_vector_norm` is required to run Vision Transformer models in ExecuTorch. Currently, model export + inference fails because ExecuTorch doesn't have a kernel for `linalg_vector_norm`.

However, there is a decomposition for the operator. Add the decomposition to the core decomp table to unblock ExecuTorch.

SS-JIA commented:
Unfortunately, it seems like the `_refs` decomp is failing a correctness check. I'm seeing errors like

```
======================================================================
ERROR: test_quick_core_backward_linalg_vector_norm_cpu_float64 (__main__.TestDecompCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ssjia/Github/pytorch/torch/testing/_internal/common_device_type.py", line 1159, in test_wrapper
    return test(*args, **kwargs)
  File "/home/ssjia/Github/pytorch/torch/testing/_internal/common_device_type.py", line 1458, in only_fn
    return fn(self, *args, **kwargs)
  File "/home/ssjia/Github/pytorch/torch/testing/_internal/common_utils.py", line 1534, in wrapper
    fn(*args, **kwargs)
  File "/home/ssjia/Github/pytorch/torch/testing/_internal/common_utils.py", line 2291, in wrapper
    fn(*args, **kwargs)
  File "/home/ssjia/Github/pytorch/test/test_decomp.py", line 589, in test_quick_core_backward
    torch.autograd.gradcheck(func, args)
  File "/home/ssjia/Github/pytorch/torch/autograd/gradcheck.py", line 2056, in gradcheck
    return _gradcheck_helper(**args)
  File "/home/ssjia/Github/pytorch/torch/autograd/gradcheck.py", line 2085, in _gradcheck_helper
    _gradcheck_real_imag(
  File "/home/ssjia/Github/pytorch/torch/autograd/gradcheck.py", line 1495, in _gradcheck_real_imag
    gradcheck_fn(
  File "/home/ssjia/Github/pytorch/torch/autograd/gradcheck.py", line 1636, in _slow_gradcheck
    raise GradcheckError(
torch.autograd.gradcheck.GradcheckError: Jacobian mismatch for output 0 with respect to input 0,
numerical:tensor([[-1.0000]], dtype=torch.float64)
analytical:tensor([[-14322.1787]], dtype=torch.float64)
```

This can be reproduced by running

```
PYTORCH_OPINFO_SAMPLE_INPUT_INDEX=0 python test/test_decomp.py TestDecompCPU.test_quick_core_backward_linalg_vector_norm_cpu_float64
```

cc: @lezcano who introduced the ref for this op in https://github.com/pytorch/pytorch/pull/78350, any idea why the test would be producing this error?

Also, @guangy10 I tried Chen's decomposition as well i.e.

```python
def decomp_linalg_vector_norm(a, order):
    # Compute the absolute values of the elements of 'a'
    abs_a = torch.abs(a)
    # Compute the sum of the absolute values raised to the power of order
    sum_p = torch.sum(torch.pow(abs_a, ord=order))
    # Compute the order-th root of the sum
    norm_value = torch.pow(sum_p, 1/order)
    return norm_value
```

But this implementation is probably not generic enough, because the same test will fail with

```
======================================================================
ERROR: test_quick_core_backward_linalg_vector_norm_cpu_float64 (__main__.TestDecompCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ssjia/Github/pytorch/torch/testing/_internal/common_device_type.py", line 1159, in test_wrapper
    return test(*args, **kwargs)
  File "/home/ssjia/Github/pytorch/torch/testing/_internal/common_device_type.py", line 1458, in only_fn
    return fn(self, *args, **kwargs)
  File "/home/ssjia/Github/pytorch/torch/testing/_internal/common_utils.py", line 1534, in wrapper
    fn(*args, **kwargs)
  File "/home/ssjia/Github/pytorch/torch/testing/_internal/common_utils.py", line 2291, in wrapper
    fn(*args, **kwargs)
  File "/home/ssjia/Github/pytorch/test/test_decomp.py", line 589, in test_quick_core_backward
    torch.autograd.gradcheck(func, args)
  File "/home/ssjia/Github/pytorch/torch/autograd/gradcheck.py", line 2056, in gradcheck
    return _gradcheck_helper(**args)
  File "/home/ssjia/Github/pytorch/torch/autograd/gradcheck.py", line 2078, in _gradcheck_helper
    func_out = func(*tupled_inputs)
  File "/home/ssjia/Github/pytorch/test/test_decomp.py", line 849, in __torch_dispatch__
    decomp_out = pytree.tree_leaves(decomposition(*args, **kwargs))
  File "/home/ssjia/Github/pytorch/torch/_prims_common/wrappers.py", line 308, in _fn
    result = fn(*args, **kwargs)
  File "/home/ssjia/Github/pytorch/torch/_refs/linalg/__init__.py", line 164, in decomp_linalg_vector_norm
    sum_p = torch.sum(torch.pow(abs_a, ord=order))
TypeError: pow() received an invalid combination of arguments - got (Tensor, ord=int), but expected one of:
 * (Tensor input, Tensor exponent, *, Tensor out = None)
 * (Number self, Tensor exponent, *, Tensor out = None)
 * (Tensor input, Number exponent, *, Tensor out = None)
 ```

cccclai commented:
How do we register custom backward function? I tried but it looks like `linalg_vector_norm_backward` is not exposed in torch aten ops
